### PR TITLE
[RLlib] Add text space serialization

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?
gym >= 0.25.0 has added text space to their standard list of spaces. Users have asked if they can use text space but their jobs would fail in the serialization step. This PR adds a forward / backward compatible serialization procedure for gym.spaces.Text class. Note that gym 0.26.0 has changed the public API for their Text so in this PR I am supporting all released gym versions.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/serialization.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 from ray.rllib.utils.spaces.repeated import Repeated
 from ray.rllib.utils.spaces.simplex import Simplex
 
+text_space_class = getattr(gym.spaces, "Text", None)
```
Comment (gjoliver) [inline rllib/utils/serialization.py:14]: I often see this pattern in our codebase:
```
try:
    from gym.spaces import Text
except ImportError:
    Text = None
```

File: rllib/utils/serialization.py [hunk 2]
```diff
@@ -115,6 +117,22 @@ def _flex_dict(sp: FlexDict) -> Dict:
             d[k] = gym_space_to_dict(s)
         return d
 
+    def _text(sp: "gym.spaces.Text") -> Dict:
```
Comment (gjoliver) [inline rllib/utils/serialization.py:120]: can then just do ``Text`` here.

**Review by gjoliver — COMMENTED** [review]
once sven's gymnasium PR is in, we will only support gym version >= 0.26 right?

**Comment by kouroshHakha** [comment]
> once sven's gymnasium PR is in, we will only support gym version >= 0.26 right?

Yes.

File: rllib/utils/serialization.py (hunk 1 above)
Comment (sven1977) [reply rllib/utils/serialization.py:14]: The preferred way is actually to do:

```
import gym

def blabla(a: gym.spaces.Text) -> gym.spaces.Text:
   ...
```

But yeah, sometimes, to make lines to become too long, we do "fall back" to `from gym.spaces import xyz`. We should probably unify this consequently across the codebase.

I prefer the solution above (simple gym import, then `gym.spaces.Text|Discrete|...`)

File: rllib/utils/serialization.py (hunk 2 above)
Comment (sven1977) [reply rllib/utils/serialization.py:120]: see my comment above

File: rllib/utils/tests/test_serialization.py [hunk 3]
```diff
@@ -140,6 +143,31 @@ def test_flex_dict(self):
         self.assertTrue(isinstance(sp["discrete"], gym.spaces.Discrete))
         self.assertTrue(isinstance(sp["tuple"], gym.spaces.Tuple))
 
+    def test_text(self):
+        # NOTE (kourosh): This unittest will automatically get activated on CI once
+        # we upgrade gym
+        if not text_space_class:
```
Comment (sven1977) [inline rllib/utils/tests/test_serialization.py:149]: nit: `is None`

**Review by sven1977 — APPROVED** [review]
LGTM. Let's merge when:
1) all CI-tests have passed
2) the new test case is validated (locally is fine) for a gym==0.25.x installation

File: rllib/utils/serialization.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/utils/serialization.py:14]: Jun's solution is more about the None assignment. But I do think we have to change this line anyway once gymansium is merged. So let's not worry about it for now and go with this? I also like `gym.spaces.Text` for consistency.

File: rllib/utils/tests/test_serialization.py (hunk 3 above)
Comment (kouroshHakha) [reply rllib/utils/tests/test_serialization.py:149]: fixed.

**Pull request merged**